File: SignIn.cs
using UnityEngine;
using System.Collections;
using UnityEngine.UI;

public class SignIn : MonoBehaviour {

    public string username;
    public string password;
    public string email;
    public InputField inputUsername;
    public InputField inputPassword;
    public InputField inputEmail;
    public Text errorTextPassword;
    public Text errorTextUsername;
    public Text errorTextEmail;
    public Text signedInText;
    public Button registerButton;
    private string formText = "";
    private string URL = "http://localhost/Unity/signin.php";

    void Start ()
    {
        errorTextPassword.enabled = false;
        errorTextUsername.enabled = false;
        errorTextEmail.enabled = false;
        signedInText.enabled = false;
    }

    public void onClickSignIn()
    {
        errorTextPassword.enabled = false;
        errorTextUsername.enabled = false;
        errorTextEmail.enabled = false;
        StopCoroutine(Register());
        StartCoroutine(Register());
    }

    IEnumerator Register()
    {
        username = inputUsername.text;
        password = inputPassword.text;
        email = inputEmail.text;
        WWWForm form = new WWWForm();
        form.AddField("username", username);
        form.AddField("password", password);
        form.AddField("email", email);
        WWW web = new WWW(URL, form);
        yield return web;
        formText = web.data;
        Debug.Log(formText);
        switch (formText)
        {
            case "signedin":

                signedInText.enabled = true;
                yield return new WaitForSeconds(2f);
                //Application.LoadLevel(4);
                Debug.Log(formText);
                break;

            case "Username already taken!":

                errorTextUsername.text = "Username already taken!";
                errorTextUsername.enabled = true;
                break;

            case "Username too short!":

                errorTextUsername.text = "Username too short! (min. 3)";
                errorTextUsername.enabled = true;
                break;

            case "Username too long!":

                errorTextUsername.text = "Username too long! (max. 20)";
                errorTextUsername.enabled = true;
                break;

            case "Email already taken!":

                errorTextEmail.text = "Email already taken!";
                errorTextEmail.enabled = true;
                break;

            case "Password too short!":

                errorTextPassword.text = "Password too short! (min. 5)";
                errorTextPassword.enabled = true;
                break;

            case "Password too long!":

                errorTextPassword.text = "Password too long! (max. 20)";
                errorTextPassword.enabled = true;
                break;

            case "Email cannot be empty!":

                errorTextEmail.text = "Email cannot be empty!";
                errorTextEmail.enabled = true;
                break;

            case "Please write correct email!":

                errorTextEmail.text = "Please write correct email!";
                errorTextEmail.enabled = true;
                break;

            default:

                Debug.Log("error in cases!");
                Debug.Log(formText);
                break;
        }
        username = "";
        password = "";
        email = "";
        StopCoroutine(Register());
    }
}

File: signin.php
<?php

		$username = mysql_real_escape_string($_POST['username']);
		$password = mysql_real_escape_string($_POST['password']);
		$email = mysql_real_escape_string($_POST['email']);
		$bool = true;
		$error = false;
		if(strlen($username)<3 && !$error)
		{
			$error = true;
			echo "Username too short!";
		}
		if(strlen($username)>20 && !$error)
		{
			$error = true;
			echo "Username too long!";
		}
		if(strlen($password)<5 && !$error)
		{
			$error = true;
			echo "Password too short!";
		}
		if(strlen($password)>20 && !$error)
		{
			$error = true;
			echo "Password too long!";
		}
		if(empty($email) && !$error)
		{
			$error = true;
			echo "Email cannot be empty!";
		}
		if(!filter_var($email, FILTER_VALIDATE_EMAIL))
		{
			$error = true;
			echo "Please write correct email!";
		}
		if(!$error)
		{
			mysql_connect("localhost", "root", "") or die(mysql_error());
			mysql_select_db("unity") or die("Cannot connect to database");
			$query = mysql_query("Select * from users");
			while($row = mysql_fetch_array($query))
			{
				$table_users = $row['username'];
				if($username == $table_users)
				{
					$bool = false;
					echo "Username already taken!";
				}
				else
				{
					if($email == $table_users)
					{
						$bool = false;
						echo "Email already taken!";
					}
				}
			}
			if($bool)
			{
				$hash = md5(rand(0,1000));
				mysql_query("INSERT INTO users (username, password, email, Account_active, Active_key) VALUES ('$username','$password', '$email', '0', '$hash')");
				echo "signedin";
				$to = $email;
				$subject = 'Verification';
				$message = '
				
	Thanks for signing up!
	Your account has been created, you can login just after you have activated your account by pressing the url below.\r\n

	------------------------
	Username: '.$username.'
	Password: '.$password.'
	------------------------
					
	Please click this link to activate your account:
	http://www.yourwebsite.com/verify.php?email='.$email.'&hash='.$hash.'
					 
	';
									 
				$headers = 'From:noreply@yourwebsite.com' . "\r\n";
				mail($to, $subject, $message, $headers);
			}
		}
?>